nil/getting-started/essentials/handling-async-execution.mdx

```diff
@@ -0,0 +1,143 @@
+# Handling async execution
+
+To support [**cross-shard communication**](../../principles#cross-shard-communication), =nil; supports smart contracts making async calls to one another directly in Solidity code. 
+
+This tutorial provides a brief theoretical and practical primer on async execution, and gives suggestions on how this feature can be integrated into an application.
+
+## Performing an async cal
+
+Consider a smart contract with the following code:
+
+```sol showLineNumbers
+import "./nil.sol";
+
+contract Caller {
+
+    using Nil for address; 
+
+    event CallCompleted(address indexed dst)
+
+    function call(address dst) public payable {
+        dst.asyncCall(address(0), gasleft(), false, msg.value, abi.encodeWithSignature("funcName"));
+        emit CallCompleted(dst);
+    }
+}
+```
+
+The `address.asyncCall()` function makes it possible for the contract to call a function in another smart contract regardless of the shard where this contract is located. There are no additional actions required for making async calls: simply adding `address.asyncCall()` and specifying its arguments is sufficient.
```

Comment:
here and further below, follow the example that I've pointed to above (use Nil.asyncCall)